### PR TITLE
feat(django): add SENTRY_SEND_PII option to allow turning it off

### PR DESCRIPTION
**Pull request by xvello**
## Problem

Add a new `SENTRY_SEND_PII` envvar and handle it for django, another PR will handle it for plugin-server. We should keep it turned on in the US region, but keep it off on EU.
It should be safe to turn on for EU's `posthog-web` pods, but let's check whether we do need it.

## Changes

Keep the default true for now, but when set to false, stop automatically collecting:

- standard sentry "client IP" field, through send_default_pii
- django access logs (info level)
- request payloads

Not having the payloads might make things harder to debug, but in that case we'll want to add explicit logging of the info we need for debugging.

### CODE REVIEW

**Comment by tiina303** [comment]
my 2 🪙  let's not make it configurable and just remove by default for everything - it will make things faster to discover if we're messing up and if we need more info we can just add it to the sentry errors and get it in the next ones.

**Comment by xvello** [comment]
> my 2 🪙 let's not make it configurable and just remove by default for everything - it will make things faster to discover if we're messing up and if we need more info we can just add it to the sentry errors and get it in the next ones.

I agree that defaulting to true is too risky, so I'm pushing a change to have it off by default. We'll set `SENTRY_SEND_PII=true` on the US region, and we could later turn it on on EU's `posthog-web` if we really need it.

I disagree with you about turning it off entirely on both regions. I'm fine with turning it off on EU with the assumption that big bugs will also affect US, where we'll still have full visibility.

File: posthog/settings/sentry.py [hunk 1]
```diff
@@ -74,17 +74,24 @@ def traces_sampler(sampling_context: dict) -> float:
...
+        sentry_logging = LoggingIntegration(level=sentry_logging_level, event_level=None)
         profiles_sample_rate = get_from_env("SENTRY_PROFILES_SAMPLE_RATE", type_cast=float, default=0.0)
 
         sentry_sdk.init(
             dsn=os.environ["SENTRY_DSN"],
             integrations=[DjangoIntegration(), CeleryIntegration(), RedisIntegration(), sentry_logging],
-            request_bodies="always",
+            request_bodies="always" if send_pii else "never",
             sample_rate=1.0,
             # Configures the sample rate for error events, in the range of 0.0 to 1.0. The default is 1.0 which means that 100% of error events are sent. If set to 0.1 only 10% of error events will be sent. Events are picked randomly.
```
Comment (pauldambra) [inline posthog/settings/sentry.py:93]: I can't figure out if the comment here refers to sample_rate or traces_sample or both, should probably push `send_default_pii` so it's below `request_bodies`

**Review by pauldambra — APPROVED** [review]
👍 

We could link out to https://github.com/getsentry/sentry-python/blob/1.18.0/sentry_sdk/scrubber.py#L17-L58 or https://docs.sentry.io/platforms/python/configuration/options/#send-default-pii

But other than that 🚢

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**